addon/models/guid-referent.js

```diff
@@ -0,0 +1,6 @@
+import OsfModel from './osf-model';
+
+/**
+ * Used only to support the polymorphic referent relationship on the Guid model
+ **/
+export default OsfModel.extend({});
```

Comment:
Spec question:

Looking at the OSF models, it several other kinds of information subclass `GuidStoredObject`. Will we need to add this base class to other models than Node/Registration, File, and User? Honest question; the definition vs usage in practice is a bit unclear.

Here is the class definition:

https://github.com/abought/osf.io/blob/723017aabd01af104ba316fa2390c575df88cd84/framework/guid/model.py#L53-L53
>     """Subclass of `StoredObject` that provisions a `Guid` for each new instance
    on save. When saving a `GuidStoredObject` for the first time, creates a new
    `Guid`, then assigns the primary key of the instance to the primary key of
    the `Guid`. Note: Subclasses should have a `StringField` primary key, since
    the key generated by the associated `Guid` will also be a string.
    """

And here is a list of things that directly inherit. Ones worth a look for ember-osf are noted in bold:
- User
- ~~Badge~~
- *NodeWikiPage*
- *Comment*
- Node

Additionally. `StoredFileNode` calls `Guid.generate` optionally in certain situations.